### PR TITLE
[ACM-2.9] ACM#10464: Extracting OpenShift release image digest

### PR DESCRIPTION
**Pull request by xenolinux**
https://issues.redhat.com/browse/ACM-10464

The peer-review approval is present in https://github.com/stolostron/rhacm-docs/pull/6392

### CODE REVIEW

File: clusters/hosted_control_planes/create_cluster_bm.adoc [hunk 1]
```diff
@@ -1,45 +1,50 @@
...
-export MACHINE_CIDR=192.168.122.0/24
-export ETCD_STORAGE="lvm-storageclass"
-export OCP_RELEASE=4.14.0-multi       
-export SSH_KEY=${HOME}/.ssh/id_rsa.pub
-oc create ns ${HOSTED_CONTROL_PLANE_NAMESPACE}
+oc create ns <hosted_cluster_namespace>-<hosted_cluster_name>
+----
+
++
+Replace `<hosted_cluster_namespace>` with your hosted cluster namespace name, for example, `clusters`. Replace `<hosted-cluster-name>` with your hosted cluster name.
```
Comment (CrystalChun) [inline clusters/hosted_control_planes/create_cluster_bm.adoc]:
```suggestion
Replace `<hosted_cluster_namespace>` with your hosted cluster namespace name, for example, `clusters`. Replace `<hosted_cluster_name>` with your hosted cluster name.
```

**Review by CrystalChun — APPROVED** [review]
/lgtm

File: clusters/hosted_control_planes/create_cluster_bm.adoc [hunk 2]
```diff
@@ -1,45 +1,50 @@
 [#creating-a-hosted-cluster-bm]
 = Creating a hosted cluster on bare metal
 
-You can create a hosted cluster or import one. For instructions to import a hosted cluster, see _Importing a hosted cluster_.
+You can create a hosted cluster on bare metal or import one. For instructions to import a hosted cluster, see _Importing a hosted cluster_.
 
 . Verify that you have a default storage class configured for your cluster. Otherwise, you might end up with pending PVCs. Enter the following commands, replacing any example variables with your information:
 
 +
 ----
```
Comment (dockerymick) [inline clusters/hosted_control_planes/create_cluster_bm.adoc:10]:
```suggestion
[source,bash]
----
```

File: clusters/hosted_control_planes/create_cluster_bm.adoc [hunk 3]
```diff
@@ -1,45 +1,50 @@
...
+----
+
++
+Replace `<hosted_cluster_namespace>` with your hosted cluster namespace name, for example, `clusters`. Replace `<hosted_cluster_name>` with your hosted cluster name.
+
+. Verify that you have a default storage class configured for your cluster. Otherwise, you might see pending PVCs. See the following example:
 
++
+----
 hcp create cluster agent \
```
Comment (dockerymick) [inline clusters/hosted_control_planes/create_cluster_bm.adoc:22]: Is the user expected to run this command in order to receive the configuration details? Or is this one command?

File: clusters/hosted_control_planes/create_cluster_bm.adoc [hunk 4]
```diff
@@ -1,45 +1,50 @@
...
+<4> Specify your base domain, for example, `krnl.es`.
+<5> Specify the etcd storage class name, for example, `lvm-storageclass`.
+<6> Specify the path to your SSH public key. The default file path is `~/.ssh/id_rsa.pub`.
+<7> Specify your hosted cluster namespace.
+<8> Specify the supported {ocp-short} version that you want to use, for example, `4.14.0-x86_64`. If you are using a disconnected environment, replace `<ocp_release_image>` with the digest image. To extract the {ocp-short} release image digest, see _Extracting the {ocp-short} release image digest_.
 
 . After a few moments, verify that your hosted control plane pods are up and running by entering the following command:
 
 +
 ----
```
Comment (dockerymick) [inline clusters/hosted_control_planes/create_cluster_bm.adoc:49]:
```suggestion
[source,bash]
----
```

File: clusters/hosted_control_planes/disconnected_intro.adoc [hunk 5]
```diff
@@ -14,16 +14,16 @@ The following diagram illustrates an example architecture of a disconnected envi
...
 
-* Key: `<registry_dns_domain_name>..<port>`
+* Key: `<registry_dns_domain_name>..<port>`, for example, `registry.hypershiftdomain.lab..5000:`. Ensure that you place `..` after the registry DNS domain name when you specify a port.
 * Value: The certificate content
 
 +
 For more information about creating a config map, see _Configuring TLS certificates for an IPv4 network_.
-. You modify the `images.config.openshift.io` custom resource (CR) specification and adds a new field named `additionalTrustedCA` with a value of `name: registry-config`.
-. You create a config map that contains two data fields. One field contains the `registries.conf` file in `RAW` format, and the other field contains the Registry CA and is named `ca-bundle.crt`. The config map belongs to the `multicluster-engine` namespace, and the config map name is referenced in other objects. For an example of a config map, see the following sample configuration:
+. Add the `additionalTrustedCA` field with a value of `name: registry-config` in the `images.config.openshift.io` custom resource (CR).
```
Comment (dockerymick) [inline clusters/hosted_control_planes/disconnected_intro.adoc]: Earlier i recall that the team agreed to not use the abbreviation for custom resource. I also remember the conversation coming up again in a meeting. Let's bring it up to the team to confirm. 
```suggestion
. Add the `additionalTrustedCA` field with a value of `name: registry-config` in the `images.config.openshift.io` custom resource.
```

File: clusters/hosted_control_planes/disconnected_intro.adoc [hunk 6]
```diff
@@ -61,38 +61,75 @@ data:
 ----
 
 . In the {mce-short} namespace, you create the `multiclusterengine` CR, which enables both the Agent and `hypershift-addon` add-ons. The {mce-short} namespace must contain the config maps to modify behavior in a disconnected deployment. The namespace also contains the `multicluster-engine`, `assisted-service`, and `hypershift-addon-manager` pods.
-. You create the objects that are necessary to deploy the hosted cluster, which include the following components:
+. Create the objects that are necessary to deploy the hosted cluster, which include the following components:
 
 ** Secrets: Secrets contain the pull secret, SSH key, and etcd encryption key.
 ** Config map: The config map contains the CA certificate of the private registry.
-** `HostedCluster`: The `HostedCluster` resource defines the configuration of the cluster that the user intends to create.
+** `HostedCluster`: The `HostedCluster` resource defines the configuration for your hosted cluster.
```
Comment (dockerymick) [inline clusters/hosted_control_planes/disconnected_intro.adoc]:
```suggestion
** `HostedCluster` resource: The `HostedCluster` resource defines the configuration for your hosted cluster.
```

File: clusters/hosted_control_planes/disconnected_intro.adoc [hunk 7]
```diff
@@ -61,38 +61,75 @@ data:
...
 ** `NodePool`: The `NodePool` resource identifies the node pool that references the machines to use for the data plane.
```
Comment (dockerymick) [inline clusters/hosted_control_planes/disconnected_intro.adoc]:
```suggestion
** `NodePool` resource: The `NodePool` resource identifies the node pool that references the machines to use for the data plane.
```

File: clusters/hosted_control_planes/disconnected_intro.adoc [hunk 8]
```diff
@@ -61,38 +61,75 @@ data:
...
+. After you create the hosted cluster objects, the HyperShift Operator creates control plane pods in the `HostedControlPlane` namespace. The `HostedControlPlane` namespace also hosts components such as Agents, bare metal hosts, and the `InfraEnv` resource.
 
-. At this point, you wait for the deployment process to be completed.
+. Create the `InfraEnv` resource. After generating an ISO image, create the bare metal hosts and their secrets that contain baseboard management controller (BMC) credentials.
+
+. The Metal3 Operator in the `openshift-machine-api` namespace inspects the new bare metal hosts.
+
+. The Metal3 Operator tries to connect and start the BMCs by using the `LiveISO` and `RootFS` values. You can specify the `LiveISO` and `RootFS` values through the `AgentServiceConfig` CR in the {mce-short} namespace.
```
Comment (dockerymick) [inline clusters/hosted_control_planes/disconnected_intro.adoc]: https://www.ibm.com/docs/en/ibm-style?topic=grammar-anthropomorphism
```suggestion
. Specify the `LiveISO` and `RootFS` values through the `AgentServiceConfig` CR in the {mce-short} namespace  to connect and start the BMCs with the Metal3 Operator.
```

File: clusters/hosted_control_planes/disconnected_intro.adoc [hunk 9]
```diff
@@ -61,38 +61,75 @@ data:
...
 
-. After the worker nodes of the `HostedCluster` resource are started, an Agent container is started. This agent establishes contact with the Assisted Service, which orchestrates the actions to complete the deployment. Initially, you need to scale the `NodePool` resource to the number of worker nodes for the `HostedCluster` resource. The Assisted Service manages the remaining tasks.
+. After you create the hosted cluster objects, the HyperShift Operator creates control plane pods in the `HostedControlPlane` namespace. The `HostedControlPlane` namespace also hosts components such as Agents, bare metal hosts, and the `InfraEnv` resource.
 
-. At this point, you wait for the deployment process to be completed.
+. Create the `InfraEnv` resource. After generating an ISO image, create the bare metal hosts and their secrets that contain baseboard management controller (BMC) credentials.
+
+. The Metal3 Operator in the `openshift-machine-api` namespace inspects the new bare metal hosts.
```
Comment (dockerymick) [inline clusters/hosted_control_planes/disconnected_intro.adoc]: This seems like it should not be a step from my perspective. What is the task that should be complete by the user

File: clusters/hosted_control_planes/disconnected_intro.adoc [hunk 10]
```diff
@@ -61,38 +61,75 @@ data:
...
+
+. The Metal3 Operator tries to connect and start the BMCs by using the `LiveISO` and `RootFS` values. You can specify the `LiveISO` and `RootFS` values through the `AgentServiceConfig` CR in the {mce-short} namespace.
+
+. After the worker nodes of the `HostedCluster` resource are ready, an Agent container starts running.
```
Comment (dockerymick) [inline clusters/hosted_control_planes/disconnected_intro.adoc]:
```suggestion
. After the worker nodes of the `HostedCluster` resource are ready, the Agent container automatically start.
```

**Review by dockerymick — CHANGES_REQUESTED** [review]

File: clusters/hosted_control_planes/disconnected_intro.adoc (hunk 9 above)
Comment (xenolinux) [reply clusters/hosted_control_planes/disconnected_intro.adoc]: Agree. Removed as a step

File: clusters/hosted_control_planes/create_cluster_bm.adoc (hunk 3 above)
Comment (xenolinux) [reply clusters/hosted_control_planes/create_cluster_bm.adoc:22]: This is a continuous one command. Added [source,bash] for this command too.

**Comment by xenolinux** [comment]
Mistakenly closed the PR.

**Pull request merged**